test/common/config/delta_subscription_impl_test.cc

```diff
@@ -1,165 +1,86 @@
+  subscription_->updateResources({"name1", "name2", "name3", "name4"});
+  expectSendMessage({}, {"name1", "name2", "name3"}, Grpc::Status::GrpcStatus::Ok, "", {});
+  subscription_->updateResources({"name4"});
 }
 
-// add/remove/add == add.
-TEST_F(DeltaSubscriptionImplTest, AddRemoveAdd) {
+TEST_F(DeltaSubscriptionImplTest, PauseBlocksRequest) {
   startSubscription({"name1", "name2", "name3"});
   subscription_->pause();
+
+  expectSendMessage({"name4"}, {"name1", "name2"}, Grpc::Status::GrpcStatus::Ok, "", {});
+  // If not for the pause, these updates would make the expectSendMessage fail due to too many
+  // messages being sent.
+  subscription_->updateResources({"name3", "name4"});
   subscription_->updateResources({"name1", "name2", "name3", "name4"});
-  subscription_->updateResources({"name1", "name2", "name3"});
+  subscription_->updateResources({"name3", "name4"});
   subscription_->updateResources({"name1", "name2", "name3", "name4"});
-  InSequence s;
-  expectSendMessage({"name4"}, {}, Grpc::Status::GrpcStatus::Ok, "", {});
-  subscription_->resume();
-}
+  subscription_->updateResources({"name3", "name4"});
 
-// remove/add/remove == remove.
-TEST_F(DeltaSubscriptionImplTest, RemoveAddRemove) {
-  startSubscription({"name1", "name2", "name3"});
-  subscription_->pause();
-  subscription_->updateResources({"name1", "name2"});
-  subscription_->updateResources({"name1", "name2", "name3"});
-  subscription_->updateResources({"name1", "name2"});
-  InSequence s;
-  expectSendMessage({}, {"name3"}, Grpc::Status::GrpcStatus::Ok, "", {});
   subscription_->resume();
 }
 
-// Starts with 1,2,3. 4 is added/removed/added. In those same updates, 1,2,3 are
-// removed/added/removed. End result should be 4 added and 1,2,3 removed.
-TEST_F(DeltaSubscriptionImplTest, BothAddAndRemove) {
+TEST_F(DeltaSubscriptionImplTest, PauseQueuesAcks) {
   startSubscription({"name1", "name2", "name3"});
   subscription_->pause();
-  subscription_->updateResources({"name4"});
-  subscription_->updateResources({"name1", "name2", "name3"});
-  subscription_->updateResources({"name4"});
-  InSequence s;
-  expectSendMessage({"name4"}, {"name1", "name2", "name3"}, Grpc::Status::GrpcStatus::Ok, "", {});
+  {
+    auto message = std::make_unique<envoy::api::v2::DeltaDiscoveryResponse>();
+    auto* resource = message->mutable_resources()->Add();
+    resource->set_name("name1");
+    resource->set_version("version1A");
+    std::string nonce = std::to_string(HashUtil::xxHash64("version1A"));
+    message->set_nonce(nonce);
+    nonce_acks_required_.push(nonce);
+    subscription_->onDiscoveryResponse(std::move(message));
+  }
+  {
+    auto message = std::make_unique<envoy::api::v2::DeltaDiscoveryResponse>();
+    auto* resource = message->mutable_resources()->Add();
+    resource->set_name("name2");
+    resource->set_version("version2A");
+    std::string nonce = std::to_string(HashUtil::xxHash64("version2A"));
```

Comment:
Nit: prefer to `const` these immutable temps.

Reply:
Done.